app/src/main/java/com/poklad/androidifystore/data/repositories/ProductsRemoteRepositoryImpl.kt

```diff
@@ -0,0 +1,35 @@
+package com.poklad.androidifystore.data.repositories
+
+import com.poklad.androidifystore.data.mapper.Mapper
+import com.poklad.androidifystore.data.mapper.ProductDtoToProductMapper
+import com.poklad.androidifystore.data.remote.StoreApi
+import com.poklad.androidifystore.data.remote.dto.ProductDtoItem
+import com.poklad.androidifystore.domain.model.ProductItem
+import com.poklad.androidifystore.domain.repositories.ProductsRepository
+
+//todo вот передать маппер єто хорошая идея или лучше создаваь обьект в лямбде? И насколько у нас тут абстракция
+// работает. Ведь мі же можем еще здесь с Бд работать, а єто новіе мапперы. ТАк чт может не передать в зависимости.
+class ProductsRemoteRepositoryImpl(
+    private val storeApi: StoreApi,
+    private val mapper: Mapper<ProductDtoItem, ProductItem>// и вот уже получаеться нельзя использовтаь Категории.
```

Comment:
if you want to pass the interface, you can provide the same interface with `@Named` annotation

Reply:
and you will have "allProductsMapper" and "productByIdMapper". But if they are the same then you can just call this mapper in the map method as I described below